lib/src/useScript/useScript.ts

```diff
@@ -1,70 +1,102 @@
-import { useEffect, useState } from 'react'
+import { useEffect, useState } from 'react';
 
-export type Status = 'idle' | 'loading' | 'ready' | 'error'
-export type ScriptElt = HTMLScriptElement | null
+export type UseScriptStatus = 'loading' | 'ok' | 'error';
 
-function useScript(src: string): Status {
-  const [status, setStatus] = useState<Status>(src ? 'loading' : 'idle')
+// Cached script statuses
+const cachedScriptStatuses: Record<string, UseScriptStatus | undefined> = {};
+
+function getScriptNode(src: string) {
+  const node: HTMLScriptElement | null = document.querySelector(
+    `script[src="${src}"]`,
+  );
+  const status = node?.getAttribute('data-status') as
+    | UseScriptStatus
+    | undefined;
+
+  return {
+    node,
+    status,
+  };
+}
+
+/**
+ * Load a script and return its status. If the script is already loaded, it will return it's load status immediately.
+ */
+export function useScript(src: string): UseScriptStatus {
+  const [status, setStatus] = useState<UseScriptStatus>(() => {
+    if (typeof window === 'undefined') {
+      // SSR Handling - always return 'loading'
+      return 'loading';
+    }
+
+    return cachedScriptStatuses[src] ?? 'loading';
+  });
 
   useEffect(
     () => {
-      if (!src) {
-        setStatus('idle')
-        return
```

Comment:
Deleted this, it felt weird to have an implicit `''` as an acceptable value, if anything I'd say we should do `string | null`